### PR TITLE
feat: get stacks

### PR DESCRIPTION
**Pull request by Onitsiky**

### CODE REVIEW

File: src/main/java/api/jcloudify/app/endpoint/rest/controller/StackController.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  @GetMapping("/users/{userId}/applications/{applicationId}/environments/{environmentId}/stacks")
+  public PagedStacksResponse getStacks(
+      @PathVariable String userId,
+      @PathVariable String applicationId,
+      @PathVariable String environmentId,
+      @RequestParam(required = false, defaultValue = "1") PageFromOne page,
+      @RequestParam(required = false, defaultValue = "10") BoundedPageSize pageSize) {
+    var data = service.findAll(applicationId, environmentId, page, pageSize);
```
Comment (Mahefaa) [inline src/main/java/api/jcloudify/app/endpoint/rest/controller/StackController.java:41]: rename service method findAll to at least findAllBy

File: src/main/java/api/jcloudify/app/endpoint/rest/controller/StackController.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  @PostMapping("/applications/{applicationId}/environments/{environmentId}/deploymentInitiation")
+  public InitiateStackDeploymentResponse initiatedStackDeployment(
+      @PathVariable String applicationId,
+      @PathVariable String environmentId,
+      @RequestBody InitiateStackDeploymentRequestBody deploymentsToInitiate) {
+    var data =
+        service.process(
+            Objects.requireNonNull(deploymentsToInitiate.getData()), applicationId, environmentId);
+    return new InitiateStackDeploymentResponse().data(data);
+  }
```
Comment (Mahefaa) [inline src/main/java/api/jcloudify/app/endpoint/rest/controller/StackController.java:60]: no /users/{userId} ?

File: src/main/java/api/jcloudify/app/endpoint/rest/controller/StackController.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  private final StackService service;
+
+  @GetMapping(
+      "/users/{userId}/applications/{applicationId}/environments/{environmentId}/stacks/{stackId}")
+  public Stack getStacks(
+      @PathVariable String userId,
+      @PathVariable String applicationId,
+      @PathVariable String environmentId,
+      @PathVariable String stackId) {
+    return service.getById(stackId);
```
Comment (Mahefaa) [inline src/main/java/api/jcloudify/app/endpoint/rest/controller/StackController.java:31]: this is TOO dependant on UUID, if the ID wasn't UUID but anything else, relying on ID only to findBy might fail.
use the userId, applicationId and environmentId

File: src/main/java/api/jcloudify/app/repository/model/Stack.java [hunk 4]
```diff
@@ -49,4 +50,10 @@ public class Stack implements Serializable {
 
   @Column(name = "id_application")
   private String applicationId;
+
+  @Column(name = "creation_datetime")
+  private Instant creationDatetime;
```
Comment (Mahefaa) [inline src/main/java/api/jcloudify/app/repository/model/Stack.java:55]: `@CreationTimestamp annotation exists, use if needed`

File: src/main/java/api/jcloudify/app/repository/model/Stack.java [hunk 5]
```diff
@@ -49,4 +50,10 @@ public class Stack implements Serializable {
...
+
+  @Column(name = "update_datetime")
```
Comment (Mahefaa) [inline src/main/java/api/jcloudify/app/repository/model/Stack.java:57]: no need to name column for a simple conversion of camelCase to snakeCase

**Review by Mahefaa — CHANGES_REQUESTED** [review]

File: src/main/java/api/jcloudify/app/endpoint/rest/controller/StackController.java (hunk 3 above)
Comment (Onitsiky) [reply src/main/java/api/jcloudify/app/endpoint/rest/controller/StackController.java:31]: I just used applicationId and environmnentId as the resource Stack is not directly linked to user but through application

File: src/main/java/api/jcloudify/app/endpoint/rest/controller/StackController.java (hunk 2 above)
Comment (Onitsiky) [reply src/main/java/api/jcloudify/app/endpoint/rest/controller/StackController.java:60]: to refactor

**Pull request merged**